### ISSUE TITLE
Feature Request: en/de/fr languages

### ISSUE THREAD

**Issue by jacques3629** [lumapu/ahoy#1479]
Good morning

Sorry I'm going through a translator

  Thanks for the software I use it every day
I find it exceptional however
Just a question
When I download the version
Install on my esp32
The version is in English
When I did an update
The version is in German
How to keep the English version
Do you plan in the future to change the language directly selectable on the configuration page

THANKS

**Comment by Peertje-62** [comment]
Hi, 
240304_ahoy_0.8.90_e5c0e8e_esp32-wroom32.bin = English
''de.bin'' is German
![Unbenannt](https://github.com/lumapu/ahoy/assets/144957788/8db3aaf1-6de4-49ed-b020-1c6f3333b964)
have a nice day

**Comment by Peertje-62** [comment]
Don't install DEV ahoy_0.8.90 has a few bugs in settings

**Comment by Gubi2023** [comment]
> Don't install DEV ahoy_0.8.90 has a few bugs in settings

see also #1480

**Comment by jacques3629** [comment]
Merci à tous 

j'ai réinstallé la version  0.8.83 qui fonctionne très bien 
pour garder la version english  je dois mettre à jour le firmware que sur les versions realease 

Thank you all

I reinstalled version 0.8.83 which works very well
to keep the english version I have to update the firmware only on the realease versions

**Comment by Peertje-62** [comment]
Hi, (I am Dutch)
The Development variant is also in English available.

Salut, (je suis néerlandais)
La variante Développement est également disponible en anglais

https://fw.ahoydtu.de/fw/
regards

**Comment by jacques3629** [comment]
Hi, I'm French, I tried opendtu I didn't really like it, a lot of problems with my official dtu-pro gateway
on the other hand, all in French
Ahoy dtu is very good even better when it is multilingual

Sincerely

**Comment by Peertje-62** [comment]
Hi,
You should use only one DTU at the time (not two together at the same time)
regards

**Comment by lumapu** [comment]
hi you're welcome to translate Ahoy to French, there are only two files which need to be translated..
For now we only have English and German versions.

**Comment by jacques3629** [comment]
@ lumapu   thank you what file, I don't know if I'm capable of it, I have no knowledge of programming
Thank you again for the essential software to fill the gaps in the official gateway which refreshes the data every 15 minutes


@Peertje_62  
ahoyddo you do 0 injection? the official gateway yes
using both I have the best of all worlds
monitoring every 5 s of 0 injection with the gateway hoymile

**Comment by Peertje-62** [comment]
Hello Jacques,
I have 3 panels 385 watts, east, south and west, each with a 300 watt inverter, this way I cover my average power consumption (during the day) & Open Fusion DTU
regards

**Comment by You69Man** [comment]
> hi you're welcome to translate Ahoy to French, there are only two files which need to be translated..
> For now we only have English and German versions.

Some times ago I've already tried to do this with GPT, it's a task for a second, and the results looked very promising. Of course a review by a native french speaker would be reasonable afterwards just to be sure of the quality.

Prompt was something like, "This is a Json file that contains strings for German and for English internationalisation (i18n). Add the french versions." 
(I don't remember exactly, but this should work)

**Comment by jacques3629** [comment]
@Peertje-62 hi, I have 21 PV 15 inverter, I manage to charge the EV
ahoy dtu manages well, open dtu is limited with so many inverters although it is also excellent
Sincerely

@You69Man gpt=chat gpt not available for me
gemini from Google tells me that the program is also available in French

Additional Information :

     AhoyDTU is manufactured by the Danish company Ahoy ApS.
     AhoyDTU is available in several languages, including French.   
     You can contact AhoyDTU support by email or telephone.

I hope this translation was useful to you.

**Comment by lumapu** [comment]
@jacques3629 no Ahoy isn't available in French for now.
This is the main language file I mentioned before:
https://github.com/lumapu/ahoy/blob/main/src/web/lang.json

The second one can be found here:
https://github.com/lumapu/ahoy/blob/main/src/web/lang.h

**Comment by You69Man** [comment]
@lumapu

Ceci est une première tentative de traduction d'Ahoy en français (site web et affichage). La traduction a été réalisée par GPT et n'est certainement pas parfaite. Elle devrait donc être contrôlée par un locuteur natif:
https://github.com/You69Man/ahoy/tree/feature/PR_display_i18n_new 

Dies ist ein erster Versuch Ahoy auf französisch zu übersetzen (Webseite sowie Display). Die Übersetzung stammt von GPT und ist sicher nicht perfekt. Sie sollte daher von einem Native Speaker kontrolliert werden:
https://github.com/You69Man/ahoy/tree/feature/PR_display_i18n_new 


Zwei Problemchen gibt es damit noch bei der Webdarstellung.
Dieses hat glaube ich irgendwas damit zu tun, dass bei den Buttons die Texte automatisch auf Großschrift konvertiert werden (CSS?) und damit die HTML-Entitäten der französischen Sonderzeichen nicht mehr funktionieren:
![grafik](https://github.com/lumapu/ahoy/assets/6373072/f7742687-5e1d-4e7b-a56f-b54b07654fac)

Und diese Texte zum "Datei auswählen" hab ich gar nicht gefunden...
![grafik](https://github.com/lumapu/ahoy/assets/6373072/20dcea7c-e6f2-4cb4-b2d4-469eb44af0f6)

Der Rest schaut schon ganz gut aus.

**Comment by lumapu** [comment]
Datei auswählen kommt vom deinem Browser, das mit den Buttons muss ich mir mal näher anschauen

**Comment by jacques3629** [comment]
hello 

hi, I added 4 panels + hms2000 inverter
with AHYO-DTU the whole works very well with NRF24L01 + CMT2300A

with the official dtu-pro gateway the HM inverter works very well in parallel with AHYO-DTU
I could do 0 injection while using AHYO-DTU + official DTU pro gateway with HM inverters

I received an official DTU WLITE S gateway for the 4 panels with the hms2000
I cannot get everything to work in parallel to make 0 injection

Does Ahoy dtu allow zero injection or will it allow it in the near future?

thank you for your work

**Comment by lumapu** [comment]
The HMS series are supported in the same way as the HM series. The HMS series has one crucial difference:
Only one DTU can communicate at a time with the inverter. Maybe you get it working with two DTUs if the adjust the communication frequencies (can be done in inverter settings).

**Comment by jacques3629** [comment]
thank you for the answer
how to know the frequency of the official hoymilles DTU WLITE S gateway that communicates with the HMS series
Google's Gemini AI tells me it communicates at 433.92 MHz
on AHYODTU I set the frequency to 860.75 MHZ (-10DBm)
that works well
What do you suggest?
THANKS

ps; I sent you a small contribution to your project, not much, just to encourage development
software

jean-Jacques

**Comment by lumapu** [comment]
As I remember correctly the original DTU communicates on 863.0 Mhz. If you first start the original DTU and then Ahoy you will only get data in Ahoy if you found the correct frequency.
(I never tried that and won't do 😊)

**Comment by jacques3629** [comment]
> Si je me souviens bien, le DTU d'origine communique sur 863,0 Mhz. Si vous démarrez d'abord le DTU d'origine, puis Ahoy, vous n'obtiendrez des données dans Ahoy que si vous avez trouvé la bonne fréquence.. (Je n'ai jamais essayé ça et je ne le ferai pas 😊)

Merci avec cette information, je vais pouvoir faire quelques tests 
chez hoymiles ça devient un peu n'importe quoi, j'ai acheté une passerelle officielle dtu-pro pour les HM après apparait de nouveaux onduleurs les hms
pas compatible avec la serie HM j'achete de nouveaux une passerelle DTU W LITE S  pour 4 panneaux HMS , mais la passerelle ne gère pas le power management 
heureusement que des logiciels comme Ahyo dtu , OPENDTU existe

**Comment by jacques3629** [comment]
![frequence-ok](https://github.com/lumapu/ahoy/assets/127342238/c2f8e97b-fa61-40a5-adc9-32d07b37a2f5)

Thanks it works with dtu-pro-s & ahoy dtu

**Comment by stefan123t** [comment]
@jacques3629 did you check the latest version 0.8.152 and the translation to french locale as in #1255 and #1778 ?
There is a [zero-export branch](https://github.com/lumapu/ahoy/tree/zero-export) in the making which currently undergoes testing.

Please double check the **Dynamic Power Limiter (DPL)** function in the step-daughter project [OpenDTU-OnBattery](https://github.com/hoylabs/OpenDTU-OnBattery/).
I have described a [possible PM+DPL setup](https://github.com/tbnobody/OpenDTU/issues/272#issuecomment-2219644352) with a Shelly Pro3EM as **Power Meter (PM)** here in issue #272.

**Comment by jacques3629** [comment]
Thanks for the software, I use the version
GIT SHA: 78cbd77 :: 0.8.152 <https://github.com/lumapu/ahoy/commits/78cbd77>

which gives me satisfaction, but things are changing, I am in the process
of changing equipment with a Deye hybrid inverter
I wish you much success in the software
